libs-scala/resources/src/main/scala/com/digitalasset/resources/ProgramResource.scala

```diff
@@ -3,44 +3,61 @@
 
 package com.digitalasset.resources
 
+import java.util.concurrent.{Executors, TimeUnit}
+
 import com.digitalasset.logging.ContextualizedLogger
 import com.digitalasset.logging.LoggingContext.newLoggingContext
 import com.digitalasset.resources.ProgramResource._
 
-import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
 import scala.concurrent.{Await, ExecutionContext}
 import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
 
-class ProgramResource[T](owner: ResourceOwner[T]) {
+class ProgramResource[T](
+    owner: ResourceOwner[T],
+    startupTimeout: FiniteDuration = 1.minute,
+    tearDownDuration: FiniteDuration = 10.seconds,
+) {
   private val logger = ContextualizedLogger.get(getClass)
 
-  def run()(implicit executionContext: ExecutionContext): Unit = {
```

Comment:
I believe this was a better approach. Why the change?

Reply:
Hmmm, maybe you're right. Currently it's gonna be the same everywhere though. Do you think we should move it back out now or wait until something needs a different implementation?

Reply:
No, your argument makes sense. I still thought it made sense to raise this.